### PR TITLE
SCANNERAPI-200 Rename sonar-scanner-api to sonar-scanner-commons

### PR DESCRIPTION
**Pull request by dbmeneses**
I'll request the repo to be renamed, and my plan would be to merge this after the rename.

### CODE REVIEW

File: batch/pom.xml [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
-    <groupId>org.sonarsource.scanner.api</groupId>
-    <artifactId>sonar-scanner-api-parent</artifactId>
+    <groupId>org.sonarsource.scanner.commons</groupId>
+    <artifactId>sonar-scanner-commons-parent</artifactId>
     <version>2.17-SNAPSHOT</version>
   </parent>
 
-  <artifactId>sonar-scanner-api-batch</artifactId>
+  <artifactId>sonar-scanner-commons-batch</artifactId>
   <name>SonarQube Scanner API - Batch</name>
```
Comment (leo-geoffroy-sonarsource) [inline batch/pom.xml:10]: any reason to keep the "API" in the name?

Comment (dbmeneses) [reply batch/pom.xml:10]: forgot that one, I'll update it, thanks

**Comment by leo-geoffroy-sonarsource** [comment]
We still have a  reference to API in NOTICE.txt, in some [headers](https://github.com/SonarSource/sonar-scanner-api/blob/ba5ea2b9a534f087a5e99493f0827017b4797bce/batch/src/main/java/org/sonarsource/scanner/api/internal/batch/BatchFactory.java#L25) , in [logs ](https://github.com/SonarSource/sonar-scanner-api/blob/ba5ea2b9a534f087a5e99493f0827017b4797bce/api/src/main/java/org/sonarsource/scanner/api/internal/JarDownloader.java#L51)

Not sure if we need to fix everything though

**Review by leo-geoffroy-sonarsource — APPROVED** [review]

**Comment by dbmeneses** [comment]
@leo-geoffroy-sonarsource I fixed the remaining things

File: .cirrus.yml [hunk 2]
```diff
@@ -20,10 +20,10 @@ env:
   ### Project variables
   DEPLOY_PULL_REQUEST: true
   ARTIFACTS: "\
-    org.sonarsource.scanner.api:sonar-scanner-api:jar,\
-    org.sonarsource.scanner.api:sonar-scanner-api-batch:jar,\
-    org.sonarsource.scanner.api:sonar-scanner-api-batch-interface:jar,\
-    org.sonarsource.scanner.api:sonar-scanner-api-parent:jar"
+    org.sonarsource.scanner.commons:sonar-scanner-commons:jar,\
+    org.sonarsource.scanner.commons:sonar-scanner-commons-batch:jar,\
```
Comment (henryju) [inline .cirrus.yml:24]: Could have been the opportunity to rename `batch` to something else. But this is really minor.

**Pull request merged**